### ISSUE TITLE
The empty shopping cart is used as an abandonedCart, which bothers me

### ISSUE THREAD

**Issue by Chenjiujiu** [piwik/piwik#20315]
When a user adds a product to the website, but then removes the product, and then leaves the site when the shopping cart is empty, it is regarded as a abandoned shopping cart，
which bothers me
My site will update the shopping cart when it is opened, which makes my site a abandoned shopping cart every time I visit it (unless he completes the order and turns it into a revenue shopping cart). I'm crazy
![图片](https://user-images.githubusercontent.com/28746442/216536922-0309e8d1-1828-4ea0-9de8-fd60770c0903.png)
![图片](https://user-images.githubusercontent.com/28746442/216537054-f8d3502e-236b-4035-9559-5956d797d3b8.png)

**Comment by sgiehl** [comment]
Hi @Chenjiujiu. What exactly would you expect from Matomo to handle this? 
Do you think we should discard shopping carts on Matomo side if they don't have a value or quantity?

**Comment by Chenjiujiu** [comment]
@sgiehl Thank you for your reply:！
I know that clearEcommerceCart() can be used to empty the shopping cart,But it can't cancel the shopping cart.
 I hope a new hook can cancel the shopping cart of matomo，Then I can send the cancel shopping cart hook when the shopping cart is empty

**Comment by webian** [comment]
@sgiehl I strumbled upon the same problem.

I think an emptied cart shouldn't be considered abandoned.

Also, Matomo logic to update the cart is "silly".
To add an item I can't simply use 'addEcommerceItem' for the product added but I have to call it for every item in the cart.
Practically I have to tell to Matomo how the cart is filled every time the user update the cart.
This is really impractical.

If the user then removes the last product that was in the cart, Matomo still think that this product is in the cart unless I call clearEcommerceCart() but that leaves an abandoned cart in the stats.

With UA and GA4 you simply need to send the updated data (product and q.ty added or removed) not the whole cart.

**Comment by sgiehl** [comment]
@webian I fully agree that this could be simplified and improved. Our product team will consider this for future improvements.

**Comment by webian** [comment]
I add that while testing a way to avoid the incorrect abandoned cart, I found that just calling trackEcommerceCartUpdate() creates an abandoned cart.

**Comment by nogamenofun98** [comment]
I also have this issue when integrating the tracking cart logic. 
I written the logic to detect if cart is empty, then call clearEcommerceCart() but I found out that it doesnt show 0 quantity in matomo there.

So I tried clear cart + trackEcommerceCartUpdate(), and this time it does show 0 quantity but it remain as abandoned cart record. I cant find any logic to totally remove the abandoned cart record. 

Is there any temporary solution to remove it? (I using self hosted on premise version)

Or at least the analytics part, exclude those abandoned cart with quantity 0 and total value 0?
![image](https://github.com/matomo-org/matomo/assets/9070825/779ce839-bc48-40ab-91c9-ba45f6beb50c)

**Comment by sgiehl** [comment]
I'm not aware a a technical solution to fully remove such abandoned carts. 
Technically it might still be an abandoned cart, as the user previously had items in his cart.
But from an analytics point of view it might indeed make sense to at least ignore such carts when generating the reports.

**Comment by webian** [comment]
In my opinion, a cart is only to be considered abandoned if it has objects inside it. If the user empties the cart, it is as if the user goes back to the initial situation when the cart was still empty and at that time the cart was not considered abandoned.

So it is not only in statistical terms that it should not be considered abandoned, but also in technical terms.

If, for example, a user places an item in the cart by mistake and then removes it, these actions should not be considered.
